proposals/stdlib/char-int-conversions.md

```diff
@@ -0,0 +1,184 @@
+ *  - The Char is one of the uppercase Latin letters 'A' through 'Z' and its [code] is less than `radix + 'A'.code - 10`. In this case, `this.code - 'A'.code + 10` is returned.
+ *  - The Char is one of the lowercase Latin letters 'a' through 'z' and its [code] is less than `radix + 'a'.code - 10`. In this case, `this.code - 'a'.code + 10` is returned.
+ */
+fun Char.digitToInt(radix: Int): Int
+
+/**
+ * Returns the numeric value of the digit that this Char represents in the specified [radix], or `null` if this Char is not a valid digit in the specified [radix].
+ * Throws an exception if the [radix] is not in the range `2..36`.
+ *
+ * A Char is considered to represent a digit in the specified [radix] if at least one of the following is true:
+ *  - [isDigit] is `true` for the Char and the Unicode decimal digit value of the character is less than the specified [radix]. In this case the decimal digit value is returned.
+ *  - The Char is one of the uppercase Latin letters 'A' through 'Z' and its [code] is less than `radix + 'A'.code - 10`. In this case, `this.code - 'A'.code + 10` is returned.
+ *  - The Char is one of the lowercase Latin letters 'a' through 'z' and its [code] is less than `radix + 'a'.code - 10`. In this case, `this.code - 'a'.code + 10` is returned.
+ */
+fun Char.digitToIntOrNull(radix: Int): Int?
+```
+
+`isDigit` is considered to be `true` for a `Char` if the Unicode general category of the `Char` is "Nd" (`CharCategory.DECIMAL_DIGIT_NUMBER`).
+
+- Introduce an extension function for `Int` to covert the non-negative single digit it represents
+to the corresponding `Char` representation.
+
+```kotlin
+
+/**
+ * Returns the Char that represents this numeric digit value in the specified [radix].
+ * Throws an exception if the [radix] is not in the range `2..36` or if this value is not less than the specified [radix].
+ *
+ * If this value is less than `10`, the decimal digit Char with code `'0'.code + this` is returned.
+ * Otherwise, the uppercase Latin letter with code `'A'.code + this - 10` is returned.
+ */
+fun Int.digitToChar(radix: Int): Char
+```
+
+We will also introduce an overload with no arguments having implicit `radix = 10`:
+- `fun Char.digitToInt(): Int`
+- `fun Char.digitToIntOrNull(): Int?`
+- `fun Int.digitToChar(): Char`
+
+Having overloads and not a default argument will be consistent with `String.toInt/toIntOrNull()` functions.
+
+## Dependencies
+
+- The correct implementation of `Char.digitToInt` relies on knowing all digit ranges among the supported
+`Char` values, see the issues [KT-30652](https://youtrack.jetbrains.com/issue/KT-30652) and [KT-39177](https://youtrack.jetbrains.com/issue/KT-39177).
+
+- Deprecating `Number.toChar()` and changing its modality will require a special support in the compiler.
+
+## Placement
+
+- module `kotlin-stdlib`
+- package `kotlin.text`
```

Comment:
`fun Char(UShort): Char`, `fun Char(Int): Char` and `Char.code` are placed in `kotlin` package

Reply:
done